core/quill.js

```diff
@@ -81,6 +82,11 @@ class Quill {
       this.disable();
     }
     this.emitter.emit(Emitter.events.READY);
+    this.emitter.on(Emitter.events.TEXT_CHANGE, (delta) => {
+      if (delta.length() > 0) {
```

Comment:
I don't think this check is necessary, TEXT_CHANGE should not fire with zero length deltas and even if it does, the consequence of resetting ql-empty is pretty small.